src/main/java/org/opensearch/ad/rest/RestValidateDetectorAction.java

```diff
@@ -1,44 +1,130 @@
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentType;
 import org.opensearch.extensions.rest.ExtensionRestRequest;
 import org.opensearch.extensions.rest.ExtensionRestResponse;
 import org.opensearch.rest.RestHandler.Route;
 import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.rest.action.RestToXContentListener;
 import org.opensearch.sdk.ExtensionRestHandler;
+import org.opensearch.sdk.ExtensionsRunner;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ad.util.RestHandlerUtils.TYPE;
+import static org.opensearch.ad.util.RestHandlerUtils.VALIDATE;
+import static org.opensearch.rest.RestStatus.*;
 
 public class RestValidateDetectorAction implements ExtensionRestHandler {
     private final Logger logger = LogManager.getLogger(RestValidateDetectorAction.class);
+    private final OpenSearchClient sdkClient;
+    private final NamedXContentRegistry xContentRegistry;
 
+    public static final Set<String> ALL_VALIDATION_ASPECTS_STRS = Arrays
+            .asList(ValidationAspect.values())
+            .stream()
+            .map(aspect -> aspect.getName())
+            .collect(Collectors.toSet());
+    public RestValidateDetectorAction(ExtensionsRunner runner, AnomalyDetectorExtension extension){
+        this.xContentRegistry = runner.getNamedXContentRegistry().getRegistry();
+        this.sdkClient = extension.getClient();
+    }
     @Override
     public List<Route> routes() {
         return List.of(new Route(POST, "/detectors/_validate"), new Route(POST, "/detectors/_validate/{type}"));
     }
 
     @Override
-    public ExtensionRestResponse handleRequest(ExtensionRestRequest request) {
+    public ExtensionRestResponse handleRequest(ExtensionRestRequest request){
         if (!EnabledSetting.isADPluginEnabled()) {
             throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
         }
-        Method method = request.method();
-
-        if (!Method.POST.equals(method)) {
-            return new ExtensionRestResponse(
-                request,
-                NOT_FOUND,
-                "Extension REST action improperly configured to handle " + request.toString()
-            );
+        AnomalyDetector detector;
+        XContentParser parser;
+        XContentBuilder builder = null;
+        ValidateAnomalyDetectorRequest validateAnomalyDetectorRequest;
+        try {
+            parser= request.contentParser(this.xContentRegistry);
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            String typesStr = validateTypeString(request);
+            DetectorValidationIssue issue=null;
+            try{
+                detector= AnomalyDetector.parse(parser);
+                //TODO Dependent on https://github.com/opensearch-project/opensearch-sdk-java/issues/211
```

Comment:
What's the dependency here?

Reply:
@dbwiddis will be following the work which is being done by you in getDetector to get an idea of how to import some parameters in validation.

Reply:
+1. The other half of validation requires more complex work with settings and searching existing detectors (e.g., won't let you create too many detectors).  Sinceo @vibrantvarun only had a short time to work on this I requested he submit what he did so far. I'll pick up [SDK #222](https://github.com/opensearch-project/opensearch-sdk-java/issues/222) next week.

Reply:
@vibrantvarun so let's not call it as a dependency on #211 in TODO

Reply:
ACK